### ISSUE TITLE
Unable to re-assign any Tickets to another Agent since v0.2.0

### ISSUE THREAD

**Issue by fishmad** [thekordy/ticketit#50]
Edit Ticket > The Agent drop down box always says: **Auto Select**
I'm unable see or select other Agents for manual ticket re-assignment.

* Have tried logged in as **'Admin who is NOT a member of category**  *he cant change it.*
* Have tried logged in as **'Admin who IS a member of category**  *he cant change it.*
* Have tried logged in as **Responsible Agent who is member of category** *he cant change it.*
* Have tried logged in as **Agent who is member of category** *he cant change it.*

![capture](https://cloud.githubusercontent.com/assets/12012101/10297720/9fb407ca-6c22-11e5-816c-b262ce51252b.JPG)

**Comment by FusionDesign** [comment]
I'm on it. Submitting a PR then checking.

**Comment by FusionDesign** [comment]
Do you have an agent in that category?

**Comment by fishmad** [comment]
Yes I have 4 agents and one is also admin.

**Comment by FusionDesign** [comment]
Can you please pull my PR and tell me if this fixes the issue? You can pull FusionDesign/master branch

**Comment by FusionDesign** [comment]
The only real changes are to Agent.php and TicketController.php

**Comment by fishmad** [comment]
I pull it, installed it, rolled back and ran migrations and seeding but still not working.

**Comment by FusionDesign** [comment]
Which branch? I can't duplicate this at all on master.

**Comment by fishmad** [comment]
So your saying its working on the master branch?

**Comment by fusiondesign** [comment]
For me it is, I'm not sure how to duplicate this issue

**Comment by fishmad** [comment]
In the master branch Models/Agent.php is causing me the issue: where('ticketit_agent', '1')->
When I remove it, it works

```
/**
 * list of all agents and returning lists array of id and name
 * @param integer $cat_id
 * @return bool
 */
public function scopeAgentsLists($query, $cat_id = null)
{
    if ($cat_id == null)
        return $query->where('ticketit_agent', '1')->lists('name', 'id')->toArray();

        return Category::find($cat_id)->agents->where('ticketit_agent', '1')->lists('name', 'id')->toArray();
}
```

//       return Category::find($cat_id)->agents->where('ticketit_agent', '1')->lists('name', 'id')->toArray();
          return Category::find($cat_id)->agents->lists('name', 'id')->toArray();    <-- Works

**Comment by fishmad** [comment]
Where is ticketit_agent located? I can't tell what's is looking for or where it's looking for it.

**Comment by thekordy** [comment]
@fishmad in Migrations/2015_07_22_123254_alter_users_table.php

``` php
Schema::table('users', function (Blueprint $table) {
    $table->boolean('ticketit_admin')->default(0);
    $table->boolean('ticketit_agent')->default(0);
});
```

And I just put 

``` php
where('ticketit_agent', '1')
```

 to make sure they are agents (refer to #33)

**Comment by fishmad** [comment]
Ah so its in the users table, so this fixed the issue of auto assigning to non agents but for me its also stopping the ability to reassign to another agent - this is effecting v0.1.0 for me

**Comment by thekordy** [comment]
I will fix that

**Comment by thekordy** [comment]
@fishmad I have made some changes, hope this resolves this issue and this one #48

**Comment by fishmad** [comment]
I can once again select agents again for ticket assignment. yay

I have tested the new master branch v0.1 and v0.2 and confirm its now working.
